contracts/interfaces/IStaticOracle.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+
+/// @title Uniswap V3 Static Oracle
+/// @notice Oracle contract for calculating price quoting against Uniswap V3
+interface IStaticOracle {
+
+    /// @notice Levels of manipulation resistance that can be used when asking for quotes
+    enum ManipulationResistance {
+        Dangerous,
+        Weak,
+        Medium,
+        Strong
+    }
+
+    /// @notice Returns the address of the Uniswap V3 factory
+    /// @dev This value is assigned during deployment and cannot be changed
+    /// @return The address of the Uniswap V3 factory
+    function factory() external view returns (address);
+
+    /// @notice Returns how many observations can be taken per minute in Uniswap V3 oracles
+    /// @dev This value is assigned during deployment and cannot be changed
+    /// @return The cardinality per minute
+    function cardinalityPerMinute() external view returns (uint8);
```

Comment:
Wondering if you could shed some light on this. an Observation is written upon the first state changing call per block, but if a number of `secondsAgo` is passed where there is no current Observation, an observation will be returned in the form that it would have been written if a block were mined at the time `secondsAgo`. Because of this, i think contextualizing observations as a fixed amount that may be able to be returned within a given minute may be confusing for developers. 

Also wondering about what exactly "taken" means here - "Returns how many observations can be taken per minute in Uniswap V3 oracles". 

additionally, using `cardinality` in this way may also be confusing, as we only use cardinality to describe the number of instances of recorded observations that may be currently stored in the oracle array. i.e., if someone grows the length of the oracle array by 2 after initialization, there are now 3 instances of observations available for storage, thus there is a cardinality of 3.

Reply:
This is not meant to be used by developers, it is meant to be used by whoever deploys the contract. It's not the same to deploy this contract in Ethereum mainnet, or in Polygon. One has blocks (approximately) every 15 seconds, and the other has blocks (approximately) every 2 seconds.

If someone were to call any of the `prepare` functions for 5 minutes, then in Ethereum mainnet, the contract would need to call `pool#increaseObservationCardinality(5 minutes / 12 seconds) = pool#increaseObservationCardinality(25)`

In Polygon, the contract would need to call `pool#increaseObservationCardinality(5 minutes / 2 seconds) = pool#increaseObservationCardinality(150)`

The idea behind `cardinalityPerMinute` is to configure how many observations are needed per minute, on the deployed chain. So that the `prepare` functions can work correctly on all chains where Uniswap is present.

Maybe I can change the docs so that they looks like this:
```
/// @notice Returns how many observations are needed per minute in Uniswap V3 oracles, on the deployed chain
/// @dev This value is assigned during deployment and cannot be changed
/// @return Number of observation that are needed per minute
```
Would that be clearer @Yorkemartin ?

Reply:
ah yes that natspec is much clearer - thanks for clarifying, I was thinking this was going to be etherum mainnet only, in the context of polygon / alt-L1's it makes sense.